sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/EFOShardSubscribersPool.java

```diff
@@ -331,6 +331,15 @@ Instant getWatermark() {
     return minTimestamp(state.values().stream().map(ShardState::getWatermark));
   }
 
+  Instant getEventTimestamp(KinesisRecord record) {
+    List<ShardState> sharedStates = new ArrayList<>(state.values());
+
+    if (sharedStates.isEmpty()) {
+      return null;
+    }
+    return sharedStates.get(0).getEventTime(record);
```

Comment:
Is it valid to get the event time from any shard state?

Reply:
We get watermark also from `ShardState#getWatermark` so it seems logical that event timestamp is also fetched from `ShardState#getEventTime`.

Reply:
SGTM